tests/integration/src/compiler_test.rs

```diff
@@ -0,0 +1,379 @@
+    pub fn expect_masm(&mut self, expected_masm_file: expect_test::ExpectFile) {
+        let program = self.ir_masm_program();
+        expected_masm_file.assert_eq(&program.to_string());
+    }
+
+    /// Get the compiled MASM as [`miden_assembly::Program`]
+    pub fn vm_masm_program(&mut self) -> miden_core::Program {
+        let assembler = Assembler::default()
+            .with_library(&StdLibrary::default())
+            .expect("Failed to load stdlib");
+        let program = self.ir_masm_program();
+        // TODO: get code map from the self.diagnotics
+        let codemap = CodeMap::new();
+        let program_ast = program.to_program_ast();
+        for module in program.modules() {
+            let core_module = module.to_module_ast(&codemap);
+            let _ = assembler
+                .compile_module(
+                    &core_module.ast,
+                    Some(&core_module.path),
+                    &mut AssemblyContext::for_module(false),
+                )
+                .expect(
+                    format!(
+                        "VM Assembler failed to compile module:\n{:?}",
+                        core_module.ast
+                    )
+                    .as_str(),
+                );
+        }
+        let core_program = assembler.compile_ast(&program_ast).unwrap();
+        core_program
+    }
+
+    /// Get the compiled MASM as [`miden_codegen_masm::Program`]
+    pub fn ir_masm_program(&mut self) -> Arc<miden_codegen_masm::Program> {
+        if self.ir_masm.is_none() {
+            let mut compiler = MasmCompiler::new(&self.session);
+            let hir = self.hir.take().expect("IR is not compiled");
+            let ir_masm = compiler.compile(hir).unwrap();
+            let frozen = ir_masm.freeze();
+            self.ir_masm = Some(frozen);
+        }
+        self.ir_masm.clone().unwrap()
+    }
+}
+
+fn compile_rust_file(rust_source: &str) -> Vec<u8> {
+    let rustc_opts = [
+        "-C",
+        "opt-level=z", // optimize for size
+        "--target",
+        "wasm32-unknown-unknown",
+    ];
+    let file_name = hash_string(&[rust_source]);
+    let temp_dir = std::env::temp_dir();
+    let input_file = temp_dir.join(format!("{file_name}.rs"));
+    let output_file = temp_dir.join(format!("{file_name}.wasm"));
+    fs::write(&input_file, rust_source).unwrap();
+    let output = Command::new("rustc")
+        .args(&rustc_opts)
+        .arg(&input_file)
+        .arg("-o")
+        .arg(&output_file)
+        .output()
+        .expect("Failed to execute rustc.");
+    if !output.status.success() {
+        eprintln!("{}", String::from_utf8_lossy(&output.stderr));
+        panic!("Rust to Wasm compilation failed!");
+    }
+    let wasm = fs::read(&output_file).unwrap();
+    fs::remove_file(&input_file).unwrap();
+    fs::remove_file(&output_file).unwrap();
+    return wasm;
+}
+
+fn wasm_to_wat(wasm_bytes: &Vec<u8>) -> String {
```

Comment:
Just out of curiosity, how easy would it be for us to add a frontend for the `.wat` format? Definitely not a priority, but might be nice to have. 

I am going to take this code though and add it to `midenc-compile` to implement support for `wat` as an output format. I had been looking for that, but thought it would be more involved than this, so this is great!

Reply:
The easiest way would be to use https://crates.io/crates/wat to convert `wat` to binary `wasm` and use the existing frontend.

Reply:
Makes sense, I'll keep that in mind for the future